File: frontend/web-content-management-front/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {User} from '../models/User.interface';



@Injectable({
  providedIn: 'root',
})
export class UserService {
  private readonly baseUrl = 'http://localhost:8081/users';

  constructor(private readonly http: HttpClient) {}

  getAllUsers(): Observable<User[]> {
    return this.http.get<User[]>(this.baseUrl);
  }

  getUserById(id: string): Observable<User> {
    return this.http.get<User>(`${this.baseUrl}/${id}`);
  }

  createUser(user: Partial<User>): Observable<User> {
    return this.http.post<User>(this.baseUrl, user);
  }

  updateUser(id: string, user: User): Observable<User> {
    return this.http.put<User>(`${this.baseUrl}/${id}`, user);
  }

  deleteUser(id: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
  isResponsible(userId: string): Observable<boolean> {
    return this.http.get<boolean>(`${this.baseUrl}/${userId}/is-responsible`);
  }
  getUsersByResponsibleId(responsibleId: string): Observable<User[]> {
    return this.http.get<User[]>(`${this.baseUrl}/responsible/${responsibleId}`);
  }
}

File: backend/web-content-management-back/src/main/java/com/example/web_content_management_back/service/UserService.java
package com.example.web_content_management_back.service;

import com.example.web_content_management_back.dto.UserDTO;

import java.util.List;

public interface UserService {
    List<UserDTO> getAllUsers();
    UserDTO getUserById(String id);
    UserDTO createUser(UserDTO userDTO);
    UserDTO updateUser(String id, UserDTO userDTO);
    void deleteUser(String id);
    boolean isResponsible(String userId);
    List<UserDTO> getUsersByResponsibleId(String responsibleId);
    List<String> getUserProjects(String userId);
    void addProjectToUser(String userId, String projectId);
    List<String> getUserDatabases(String userId);;
    void addDatabaseToUser(String userId, String databaseId);
}